Prepare v0.6.0 (#1434)
ManimCommunity/manim: docs/source/changelog/0.6.0-changelog.rst

******
v0.6.0
******

:Date: May 01, 2021

Contributors
============

A total of 42 people contributed to this
release. People with a '+' by their names authored a patch for the first
time.

* Abel Aebker
* Abhijith Muthyala
* Adam Ryczkowski +
* Alex Lembcke +

> * Anton Ballmaier +

Reviewer: ```suggestion
* Anton Ballmaier
```

I have authored a patch in 0.5 as well. Other names may be wrong as well?

Reviewer 2: Can you link the PR, this may help to debug the dev_changelog.py script.

Reviewer 2: Nvm, I found it: 
![image](https://user-images.githubusercontent.com/29168469/116787067-e2911380-aa56-11eb-997b-fd615f6e54c1.png)

The issue here is that your name changed between v0.4.0 -> v0.5.0 and v0.5.0 -> v0.6.0  from `AntonBallmaier` to `Anton Ballmaier` with a space -- our script works by pulling the author name and comparing author names from the set of past contributors to current contributors. I suppose we could do it based on a combination of email/user

   :meth:`~.Scene.interactive_embed` allows interaction with Scene via mouse and keyboard as well as dynamic commands via an iPython terminal.

* `#1261 <https://github.com/ManimCommunity/manim/pull/1261>`__: Render image automatically if no animation is played in a scene

Enhancements
------------

* `#1398 <https://github.com/ManimCommunity/manim/pull/1398>`__: Fix and enhance `Mobject.arrange_in_grid`
   `arrange_in_grid` now actually arranges submobjects in a grid. Added new parameters `buff`, `cell_alignment`, `row_alignments`, `col_alignments`, `row_heights`, `col_widths`, `flow_order`.

* `#1407 <https://github.com/ManimCommunity/manim/pull/1407>`__: Fix bug in :meth:`vector_coordinate_label` and move it to :class:`~.geometry.py`

* `#1380 <https://github.com/ManimCommunity/manim/pull/1380>`__: Allow image objects as background images

* `#1391 <https://github.com/ManimCommunity/manim/pull/1391>`__: Add `path_arc` support to `.animate` syntax
   The parameter `path_arc` of :class:`~.Transform` now works with the `.animate` syntax

* `#1364 <https://github.com/ManimCommunity/manim/pull/1364>`__: Added :meth:`~.Mobject.match_points`

>    - Added :func:`~.Mobject.become_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.

Reviewer 3: CC @Darylgolden 
```suggestion
   - Added :meth:`~.Mobject.match_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.
```

Reviewer 4: LGTM

Reviewer 5: can you change it in the PR title, then behackl would render this again?

Author: I've already taken care of it. 👍

* `#1320 <https://github.com/ManimCommunity/manim/pull/1320>`__: Add `jpeg` extension to the default image extensions

* `#1234 <https://github.com/ManimCommunity/manim/pull/1234>`__: New :class:`~.Mobject` method, :meth:`~.Mobject.get_midpoint`
   Easily get the middle point of a line of an :class:`~.Mobject` with :meth:`~.Mobject.get_midpoint`. (still needs to be adjusted)

* `#1237 <https://github.com/ManimCommunity/manim/pull/1237>`__: Notify user if they are using an outdated version of Manim

* `#1308 <https://github.com/ManimCommunity/manim/pull/1308>`__: Improved ManimBanner animations

* `#1275 <https://github.com/ManimCommunity/manim/pull/1275>`__: Add SVG <line> element support to :class:`~.SVGMobject`

* `#1238 <https://github.com/ManimCommunity/manim/pull/1238>`__: Add parameter `about_point` for `rotate`
   `about_point` parameter for `rotate`

* `#1260 <https://github.com/ManimCommunity/manim/pull/1260>`__: Change Brace from Tex to SVG (#1258)

* `#1122 <https://github.com/ManimCommunity/manim/pull/1122>`__: Support for specifying the interpolation algorithms for individual ImageMobjects

> * `#1283 <https://github.com/ManimCommunity/manim/pull/1283>`__: Set default value of keyword ``random_seed`` in Scene to None (was 0 and fixed before)

Reviewer: ```suggestion
* `#1283 <https://github.com/ManimCommunity/manim/pull/1283>`__: Set default value of keyword ``random_seed`` in Scene to ``None`` (was 0 and fixed before)
```

Reviewer 2: Fixed the title and also added a reference to :class:`~.Scene`

* `#1220 <https://github.com/ManimCommunity/manim/pull/1220>`__: Added sanity checks to :meth:`~.Mobject.add_to_back` for Mobjects
   Add Mobject `add_to_back` sanity checks:
   - Raises ValueError when Mobject tries to add itself
   - Raises TypeError when a non-Mobject is added
   - Filters out incoming duplicate submobjects if at least one instance of that submobject exists in the list

> * `#1249 <https://github.com/ManimCommunity/manim/pull/1249>`__: Set corners of :class:`~.Rectangle` in counterclockwise direction
>    Set the corners of `Rectangle` in counterclockwise direction to be consistent with other geometry `Mobject`s. `Transform`ations between `Rectangle` and other geometry `Mobject`s look better now.

Reviewer: ```suggestion
* `#1249 <https://github.com/ManimCommunity/manim/pull/1249>`__: Set corners of :class:`~.Rectangle` in counterclockwise direction to be consistent with other geometry `Mobject`s. `Transform`ations between `Rectangle` and other geometry `Mobject`s look better now.
```

Reviewer: It sound a bit repetitive.

Reviewer 2: I modified the body of this: https://github.com/ManimCommunity/manim/pull/1249
@abhi-6988 @abhi3019  Please review and modify if you find the language undescriptive.

Reviewer 3: > I modified the body of this: #1249

Looks good, no issues with that thanks! I've also updated the title and changelog of #1385, please have a look

******
v0.6.0
******

:Date: May 01, 2021

Contributors
============

A total of 42 people contributed to this
release. People with a '+' by their names authored a patch for the first
time.

* Abel Aebker
* Abhijith Muthyala
* Adam Ryczkowski +
* Alex Lembcke +
* Anton Ballmaier +
* Aron

> * Aron Fischer

Reviewer 5: I think both the names are the same right? @cobordism

* Alex Lembcke +
* Anton Ballmaier
* Aron
* Benjamin Hackl
* Darylgolden
* Deniz Hasler +
* Devin Neal
* Elisha Hollander +
* Erik Tastepe +
* Jan-Hendrik Müller
* Jason Villanueva
* Laith Bahodi
* Mark Miller
* Mohammad Al-Fetyani
* Naveen M K
* Newell Jensen +
* Nidhal Baccouri +
* Nikhil Garuda +
* Peilonrayz +
* Raghav Goel
* Ricky Chon +
* friedkeenan
* kamilczerwinski22 +
* sparshg

The patches included in this release have been reviewed by
the following contributors.

* Aathish Sivasubrahmanian
* Abel Aebker
* Abhijith Muthyala
* Adam Ryczkowski
* Alex Lembcke
* Anton Ballmaier
* Aron
* Benjamin Hackl
* Darylgolden
* Deniz Hasler
* Devin Neal
* Elisha Hollander
* GameDungeon
* Hugues Devimeux
* Jan-Hendrik Müller
* Jason Villanueva
* KingWampy
* Laith Bahodi
* Mark Miller
* Mohammad Al-Fetyani
* Naveen M K
* Nidhal Baccouri
* Nikhil Garuda
* Oliver
* Philipp Imhof
* Raghav Goel
* Ricky Chon
* friedkeenan
* sparshg

Pull requests merged
====================

A total of 111 pull requests were merged for this release.

Breaking changes
----------------

* `#1347 <https://github.com/ManimCommunity/manim/pull/1347>`__: Restructure vector_field module and add documentation

>    `VectorField` is renamed to `ArrowVectorField` and a new class `VectorField` is added as a superclass for `ArrowVectorField` and `StreamLines`. Also the class `AnimatedStreamLines` is removed. It's functionality is moved to `StreamLines`. Added a lot of new options when working with vector fields. `ShowPassingFlashWithThinningStrokeWidth` was moved to the indication module.

Reviewer 2: ```suggestion
   :class`~.VectorField` is renamed to :class:`~.ArrowVectorField` and a new class :class:`~.VectorField` is added as a superclass for :class:`~.ArrowVectorField` and :class:`~.StreamLines`. `AnimatedStreamLines` is removed. It's functionality is moved to :class:`~.StreamLines`. Added a lot of new options when working with vector fields. :class:`~.ShowPassingFlashWithThinningStrokeWidth` was moved to the indication module.
```

   - Removed the ``--background_color`` option. Reassigned the ``--background_color`` option's shorthand ``-c`` to ``--config_file``.
   - Removed the ``--leave_progress_bars`` option. Use ``--progress_bars=leave`` instead.
   - Removed the deprecated render quality flags, in particular: ``-l``, ``-m``, ``-h``, ``-k``.
   - Removed the ``--sound`` option. It lost support long ago with the removal of SoX.

Deprecated classes and functions
--------------------------------

* `#1354 <https://github.com/ManimCommunity/manim/pull/1354>`__: Refactored a few functions in space_ops.py, deprecated :func:`~.angle_between`

* `#1370 <https://github.com/ManimCommunity/manim/pull/1370>`__: Remove TexMobject and TextMobject
   TexMobject and TextMobject have been deprecated for a while, they are now fully removed. Use Tex or MathTex instead.

* `#1349 <https://github.com/ManimCommunity/manim/pull/1349>`__: Removed the deprecated ``SmallDot`` mobject

* `#1259 <https://github.com/ManimCommunity/manim/pull/1259>`__: Removed deprecated CairoText class

>    - Removed `CairoText` class from `manim.mobject.text_mobject`
>     - Removed `'CairoText'` from `__all__`
>     - Removed reference to `CairoText` in docstring

Reviewer 2: This indentation needs to be fixed, but I think this s3ection is honestly pretty redundant:
```suggestion

```
